runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/util/BoundedQueueExecutor.java

```diff
@@ -224,9 +224,10 @@ private void executeMonitorHeld(Runnable work, long workBytes) {
           () -> {
             String threadName = Thread.currentThread().getName();
             try {
-              if (work instanceof Work) {
+              if (work instanceof ExecutableWork) {
                 String workToken =
-                    String.format("%016x", ((Work) work).getWorkItem().getWorkToken());
+                    String.format(
+                        "%016x", ((ExecutableWork) work).work().getWorkItem().getWorkToken());
                 Thread.currentThread().setName(threadName + ":" + workToken);
```

Comment:
could you add a test where you schedule a ExecutableWork and then inside it verify the current thread name contains the work token?

Reply:
done